rust/agama-lib/src/software/client.rs

```diff
@@ -189,6 +191,28 @@ impl<'a> SoftwareClient<'a> {
         }
     }
 
+    /// Selects packages by user
+    ///
+    /// Adds the given packages to the proposal.
+    ///
+    /// * `names`: package names.
+    pub async fn select_packages(&self, names: Vec<String>) -> Result<(), ServiceError> {
+        let names: Vec<_> = names.iter().map(|n| n.as_ref()).collect();
+        self.set_resolvables(
+            USER_RESOLVABLES_LIST,
+            ResolvableType::Package,
+            names.as_slice(),
+            true,
+        )
+        .await?;
+        Ok(())
+    }
+
+    pub async fn user_selected_packages(&self) -> Result<Vec<String>, ServiceError> {
```

Comment:
why not just selected_packages?

Reply:
Because they are selected by the user, not by the product definition itself. And it is consistent with `user_selected_patterns`.

Reply:
Ok, there is already user_selected_patterns and selected_patterns methods and this is the equivalent for the first one.